### ISSUE TITLE
No possible successful shutdown for --fast-and-loose

### ISSUE THREAD

**Issue by tsutsu** [paritytech/parity#7910]
I'm running:

 - **Which Parity version?**: `Parity/v1.10.0-unstable-f8bf7e7d4-20180118/x86_64-macos/rustc1.23.0`
 - **Which operating system?**: MacOS
 - **How installed?**: from source
 - **Are you fully synchronized?**: no
 - **Which network are you connected to?**: ropsten
 - **Did you try to restart the node?**: yes (*that's the problem!*)

Quoting `parity --help` (emphasis mine):

> `--fast-and-loose`
> Disables DB WAL, which gives a significant speed up but means an **unclean** exit is unrecoverable.

Expected behaviour: that there is actually such a thing as a "clean" exit, that won't corrupt my DB when using `--fast-and-loose`.

Actual behaviour:

* the database is corrupted when you press `Ctrl+C`.
* the database is corrupted when you send `killall -HUP parity`.
* the database is corrupted even if I ask very nicely :(

Looking at the source, there's no cleanup-DB-on-quit code in either [parity/main.rs](https://github.com/paritytech/parity/blob/master/parity/main.rs) or [parity/run.rs](https://github.com/paritytech/parity/blob/master/parity/run.rs). And, in fact, [ethcore/src/blockchain/blockchain.rs](https://github.com/paritytech/parity/blob/master/ethcore/src/blockchain/blockchain.rs) never even calls anything like `KeyValueDB::close()`.

**Comment by Aggour** [comment]
The --fast-and-loose argument doesn't allow for a clean exit on purpose. It's a feature meant to be used for quick synchronization on testnet and not for use on the production network.

**Comment by andresilva** [comment]
The reason you don't see any call like `KeyValueDB::close()` is because the database is closed automatically when it is dropped, that's done [here](https://github.com/paritytech/rust-rocksdb/blob/master/src/rocksdb.rs#L831). We fixed an issue recently where some values weren't being dropped correctly on shutdown (https://github.com/paritytech/parity/pull/7695) and I think your build doesn't include that. I just ran with `--fast-and-loose` and exited a couple of times with ctrl-c and there was no db corruption. Try a recent nightly or build the latest master and check if it still happens.
